consensus/state.go

```diff
@@ -100,14 +100,21 @@ func (consensus *Consensus) getNeighborConsensusState(neighbor *node.RemoteNode)
 	return replyMsg, nil
 }
 
-// getAllNeighborsConsensusState returns the latest block info of all neighbors
-// by calling getNeighborConsensusState on all of them concurrently.
-func (consensus *Consensus) getAllNeighborsConsensusState() (map[string]*pb.GetConsensusStateReply, error) {
-	allNeighbors := consensus.localNode.GetVotingNeighbors(nil)
-	if len(allNeighbors) < minConsensusStateNeighbors {
-		allNeighbors = consensus.localNode.GetNeighbors(nil)
+// getVotingNeighborsConsensusState returns the latest block info of voting
+// neighbors by calling getNeighborConsensusState on all of them concurrently.
+// It will also update the consensus state of non-voting neighbors.
+func (consensus *Consensus) getVotingNeighborsConsensusState() (map[string]*pb.GetConsensusStateReply, error) {
+	allNeighbors := consensus.localNode.GetNeighbors(nil)
+	votingNeighbors := consensus.localNode.GetVotingNeighbors(nil)
+	if len(votingNeighbors) < minConsensusStateNeighbors {
+		votingNeighbors = allNeighbors
 	}
-	allStates := make(map[string]*pb.GetConsensusStateReply, len(allNeighbors))
+	votingNeighborsMap := make(map[string]struct{}, len(votingNeighbors))
+	for _, neighbor := range votingNeighbors {
+		votingNeighborsMap[neighbor.GetID()] = struct{}{}
```

Comment:
Why we need to initialize `votingNeighborsMap` , we may have key with empty value, which previous code wont have

Reply:
This is map not slice, make will not create empty value